### PR TITLE
feat(ui5-combobox): add suggestions grouping

### PR DESCRIPTION
**Pull request by ndeshev**
Enable users to set items to groups. Groups are visually identified in the suggestions list by group headers. 

For group headers a new component is introduced - ComboBoxGroupItem. The grouping is based on the items order as they are declared in the markup - the items between two group headers are considered to belong to the first one.

Example - in the code below - the items between the "A" group item and the "Random Group Item" are grouped to the "A" group while the last item belongs to the "Random Group Item Text" group.

```
        <ui5-combobox id="combo-grouping">
		<ui5-cb-group-item text="A"></ui5-cb-group-item>
		<ui5-cb-item text="Algeria"></ui5-cb-item>
		<ui5-cb-item text="Argentina"></ui5-cb-item>
		<ui5-cb-item text="Australia"></ui5-cb-item>
		<ui5-cb-item text="Austria"></ui5-cb-item>

		<ui5-cb-group-item text="Random Group Item Text"></ui5-cb-group-item>
		<ui5-cb-item text="Albania"></ui5-cb-item>
	</ui5-combobox>
```

cb-grouping

Fixes: #3371

### CODE REVIEW

File: packages/main/src/ComboBoxGroupItem.js
Comment (fifoosid) [inline packages/main/src/ComboBoxGroupItem.js:56]: ```isGroupItem``` fits better, because it returns a boolean

File: packages/main/src/ComboBox.js
Comment (fifoosid) [inline packages/main/src/ComboBox.js:687]: You can just use ```idx```, ```nextItemIdx``` is not needed

**Review by fifoosid — CHANGES_REQUESTED** [review]
Add ```ui5-cb-group-item``` to the appenddocs tag in ```ComboBox.js```
Add a sample in ```ComboBox.sample.html```

File: packages/main/src/ComboBox.js
Comment (MapTo0) [inline packages/main/src/ComboBox.js:733]: just `=> item.focused` its a predicate anyway

**Review by MapTo0 — COMMENTED** [review]
I suggest we deep test this changed once the value change pull request get merged

**Review by fifoosid — APPROVED** [review]
LGTM now, someone from @SAP/ui5-webcomponents-topic-rl  should approve and we can merge it

File: packages/main/src/ComboBox.js
Comment (MapTo0) [inline packages/main/src/ComboBox.js:657]: that reads hard

**Review by MapTo0 — DISMISSED** [review]
/ ---------- /

1. type "b"
2. arrow down - b should be removed

/ ---------- /

1. Type "a"
2. items gets filtered, but not autocompleted

**Comment by MapTo0** [comment]
Actually, there is a double focus all the time :D

**Comment by ndeshev** [comment]
There is no double focus for me?
When a group item is reached with the keyboard the typed in value should not be cleared, as it is in UI5. 
The autocomplete also is working fine, this is strange. please pull the last changes, something is not right.

**Comment by hristop** [comment]
lgtm, I think we can merge when someone with rights approve the change

**Review by ilhan007 — APPROVED** [review]

**Pull request merged**
